vcd/resource_vcd_network.go

```diff
@@ -187,25 +208,28 @@ func resourceVcdNetworkCreate(d *schema.ResourceData, meta interface{}) error {
 			},
 			BackwardCompatibilityMode: true,
 		},
-		EdgeGateway: &types.Reference{
-			HREF: edgeGateway.EdgeGateway.HREF,
-		},
 		IsShared: d.Get("shared").(bool),
 	}
 
-	log.Printf("[INFO] NETWORK: %#v", newnetwork)
+	orgVDCNetwork.EdgeGateway = &types.Reference{
+		HREF: edgeGateway.EdgeGateway.HREF,
+	}
+
+	util.Logger.Printf("[INFO] NETWORK: %#v", orgVDCNetwork)
+	util.Logger.Printf("[INFO] NETWORK: %#v", orgVDCNetwork.Configuration)
+	util.Logger.Printf("[INFO] NETWORK: %#v", orgVDCNetwork.Configuration.IPScopes)
 
 	err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
-		return resource.RetryableError(vdc.CreateOrgVDCNetwork(newnetwork))
+		return resource.RetryableError(vdc.CreateOrgVDCNetworkWait(orgVDCNetwork))
 	})
 	if err != nil {
-		return fmt.Errorf("Error: %#v", err)
+		return fmt.Errorf("error: %#v", err)
 	}
 
-	err = vdc.Refresh()
-	if err != nil {
-		return fmt.Errorf("error refreshing VDC: %#v", err)
-	}
+	// err = vdc.Refresh()
```

Comment:
Why Refresh() is not needed here anymore?

Reply:
Because `vdc.FindVDCNetwork` already includes it.

Reply:
So shouldn't we remove commented out code?